lib/src/kbasesearchengine/main/DecorateWithNarrativeInfo.java

```diff
@@ -0,0 +1,121 @@
+package kbasesearchengine.main;
+
+import kbasesearchengine.*;
+import kbasesearchengine.ObjectData;
+import kbasesearchengine.common.GUID;
+import us.kbase.common.service.JsonClientException;
+import kbasesearchengine.events.handler.CloneableWorkspaceClientImpl;
+import kbasesearchengine.events.handler.EventHandler;
+import kbasesearchengine.events.handler.WorkspaceEventHandler;
+import us.kbase.common.service.Tuple5;
+import us.kbase.common.service.Tuple9;
+import workspace.*;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+
+/**
+ * Created by umaganapathy on 1/26/18.
+ */
+public class DecorateWithNarrativeInfo implements SearchInterface {
+
+    /** The storage code for workspace events. */
+    public static final String STORAGE_CODE = "WS";
+    private final WorkspaceClient wsClient;
+    private final SearchInterface searchInterface;
+
+    public DecorateWithNarrativeInfo(SearchInterface searchInterface,
+                                     WorkspaceClient wsClient) {
+        this.searchInterface = searchInterface;
+        this.wsClient = wsClient;
+    }
+
+    public SearchTypesOutput searchTypes(SearchTypesInput params, String user)
```

Comment:
final vars - looks like a lot of these so not pointing out any more

Reply:
Have added.